### PR TITLE
Menu reduction

### PR DESCRIPTION
**Pull request by Trenly**
@OfficialEsco This should resolve these three comments I made in #12 
- Handles issues which contain a `#` in them or which are from other repositories (e.g `trenly/winget-pkgs#12`)
- Checks issue validity by seeing if the URL returns a valid response code
- Reduces duplicate code by moving the showing of the menu outside of the switch for the PrBodyContentReply

I've done some light testing, see microsoft/winget-pkgs#24579
image

(I'll look into the parameters issue you mentioned separately)

### CODE REVIEW

File: Tools/YamlCreate.ps1 [hunk 1]
```diff
@@ -872,99 +890,85 @@ Function Test-Manifest {
...
-                switch ( KeypressMenu -Prompt $_menu["Prompt"] -Entries $_menu["Entries"] -DefaultString $_menu["DefaultString"] -HelpText $_menu["HelpText"]) {
-                    'Y' { $PrBodyContentReply += @($_line.Replace("[ ]", "[X]")) }
-                    default { $PrBodyContentReply += @($_line) }
-                }
             }
     
             '*winget validate*' {
                 if ($?) {
-                    $PrBodyContentReply += @($_line.Replace("[ ]", "[X]"))
+                    $PrBodyContentReply += $_line.Replace("[ ]", "[X]"), "`n"
```
Comment (OfficialEsco) [inline Tools/YamlCreate.ps1:917]:
```suggestion
                    $PrBodyContentReply += @($_line.Replace("[ ]", "[X]"))
```

File: Tools/YamlCreate.ps1 [hunk 2]
```diff
@@ -872,99 +890,85 @@ Function Test-Manifest {
...
-                        'Y' { $PrBodyContentReply += @($_line.Replace("[ ]", "[X]")) }
-                        default { $PrBodyContentReply += @($_line) }
-                    }
                 }
             }
     
             '*tested your manifest*' {
                 if ($script:SandboxTest -eq '0') {
-                    $PrBodyContentReply += @($_line.Replace("[ ]", "[X]"))
+                    $PrBodyContentReply += $_line.Replace("[ ]", "[X]"), "`n"
```
Comment (OfficialEsco) [inline Tools/YamlCreate.ps1:933]:
```suggestion
                    $PrBodyContentReply += @($_line.Replace("[ ]", "[X]"))
```

File: Tools/YamlCreate.ps1 [hunk 3]
```diff
@@ -872,99 +890,85 @@ Function Test-Manifest {
...
+        }
 
-                switch ( KeypressMenu -Prompt $_menu["Prompt"] -Entries $_menu["Entries"] -DefaultString $_menu["DefaultString"]) {
-                    'Y' { $PrBodyContentReply += @($_line.Replace("[ ]", "[X]")) }
-                    default { $PrBodyContentReply += @($_line) }
-                }
+        if ($_showMenu) {
+            switch ( KeypressMenu -Prompt $_menu["Prompt"] -Entries $_menu["Entries"] -DefaultString $_menu["DefaultString"] -HelpText $_menu["HelpText"] -HelpTextColor $_menu["HelpTextColor"]) {
+                'Y' { $PrBodyContentReply += $_line.Replace("[ ]", "[X]"), "`n" }
+                default { $PrBodyContentReply += $_line, "`n" }
```
Comment (OfficialEsco) [inline Tools/YamlCreate.ps1:971]:
```suggestion
                'Y' { $PrBodyContentReply += @($_line.Replace("[ ]", "[X]")) }
                default { $PrBodyContentReply += @($_line) }
```

**Review by OfficialEsco — COMMENTED** [review]
Uhhm are these unintended reverts? 😬

**Comment by Trenly** [comment]
> Uhhm are these unintended reverts? 😬

Yes, yes they are

**Review by OfficialEsco — APPROVED** [review]
Can't really test the Issue part, but the PR didnt break anything so

**Pull request merged**